core/server/src/main/java/alluxio/master/journal/JournalWriter.java

```diff
@@ -135,22 +135,25 @@ public synchronized JournalOutputStream getCheckpointOutputStream(long latestSeq
     return mCheckpointOutputStream;
   }
 
-  /**
-   * Returns an output stream for the journal entries. The returned output stream is a singleton for
-   * this writer.
-   *
-   * @return the output stream for the journal entries
-   * @throws IOException if an I/O error occurs
-   */
-  public synchronized JournalOutputStream getEntryOutputStream() throws IOException {
+  public synchronized void writeEntry(JournalEntry entry) throws IOException {
     if (mCheckpointOutputStream == null || !mCheckpointOutputStream.isClosed()) {
       throw new IOException("The checkpoint must be written and closed before writing entries.");
     }
     if (mEntryOutputStream == null) {
       mEntryOutputStream = new EntryOutputStream(mUfs, mJournal.getCurrentLogFilePath(),
           mJournal.getJournalFormatter(), this);
     }
-    return mEntryOutputStream;
+    mEntryOutputStream.writeEntry(entry);
+  }
+
+  public synchronized void flushEntry() throws IOException {
```

Comment:
I think this should just be called `flush`, since this is not associated with a particular entry.